packages/block-editor/src/store/selectors.js

```diff
@@ -1595,7 +1595,7 @@ const isBlockVisibleInTheInserter = (
 	checkedBlocks.add( blockName );
 
 	// If parent blocks are not visible, child blocks should be hidden too.
-	if ( !! blockType.parent?.length ) {
+	if ( Array.isArray( blockType?.parent ) ) {
```

Comment:
O think we should cast to array instead, otherwise, the check won't be applied if there's a single string parent.

Reply:
Can `parent` be a string? It doesn't seem so according to [the docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#parent). Not knowing all the details of this, I'm a bit reluctant to cast the type.

Reply:
I doubt it would work with a single string value correctly.

In the `block.json` schema there is also an array enforced:

https://github.com/WordPress/gutenberg/blob/fa120809525a00ee4cd49c88e15f4520fc9c9d04/schemas/json/block.json#L65

Reply:
After reading a bit. I’m inclined to follow the advice from @youknowriad and cast string values to array to make it backward compatible if there are existing blocks that used the string dur the lack of validation.

Reply:
Pushed https://github.com/WordPress/gutenberg/pull/66234/commits/a6425e64ea13ec3059d901ce6d526c430398c718

Reply:
I've also prepared https://github.com/WordPress/gutenberg/pull/66250 to discuss a potential better fix. That would need more testing, so I'd appreciate if we can ship this PR (hotfix) while we work on #66250.